test/controllers/foreman/organizations_controller_test.rb

```diff
@@ -29,4 +29,54 @@ def test_use_newly_created_org
     assert_redirected_to step2_organization_path(:id => org.to_param)
     assert_equal session[:organization_id], org.id
   end
+
+  def test_simple_content_access_disable
+    # use org with SCA enabled
+    org = get_organization(:organization2)
+    Organization.any_instance.stubs(:simple_content_access?).returns true
+    # assert correct task will be started
+    @controller.expects(:async_task).with(::Actions::Katello::Organization::SimpleContentAccess::Disable, org.id.to_s)
+    # send request to disable SCA
+    put :update, params: { id: org.id, simple_content_access: false }
+  end
+
+  def test_simple_content_access_enable
+    # use org with SCA disabled
+    org = get_organization(:organization2)
+    Organization.any_instance.stubs(:simple_content_access?).returns false
+    # assert correct task will be started
+    @controller.expects(:async_task).with(::Actions::Katello::Organization::SimpleContentAccess::Enable, org.id.to_s)
+    # send request to enable SCA
+    put :update, params: { id: org.id, simple_content_access: true }
+  end
+
+  def test_simple_content_access_unchanged
+    # use org with SCA enabled
+    org = get_organization(:organization2)
+    Organization.any_instance.stubs(:simple_content_access?).returns true
+    # assert SCA task was not initiated
+    @controller.expects(:async_task).never
+    # update org and don't change SCA
+    put :update, params: { id: org.id, simple_content_access: true }
+  end
+
+  def test_edit_override_can_toggle
+    org = get_organization(:organization2)
+    Organization.any_instance.stubs(:service_level)
+    Organization.any_instance.stubs(:service_levels).returns []
+    Katello::UpstreamConnectionChecker.any_instance.expects(:can_connect?).returns true
+    Organization.any_instance.expects(:simple_content_access?).returns true
+    get :edit, params: { id: org.id }
+    assert_instance_variable_set :can_toggle_sca, true
```

Comment:
Because you have an expectation that the org received `simple_content_access?` this assertion is redundant since that method wouldn't have been called were `@can_toggle_sca` false. IMO, remove this assertion and add an another checking that the response was successful.

Also, checking instance variables is unusual. Better to test the response body to see how it was affected, but I don't think that's necessary here because the `simple_content_access` assertion does the trick  :)

Reply:
Checking the response body feels a bit like testing the view to me, but this is `controller_test` so we should be testing the controller.

Also I realized that `@response.body` won't contain what we would expect anyway, because our overrides are inserted using Deface after the response has already been sent.

I decided on testing the instance variable since really the test should only be testing the code in the `edit` method, and setting the variable is all that code really does.   In light of this I feel like it makes sense to keep this assertion.  Thoughts?

Reply:
Testing how that instance variable influences the response is what's important and your expectations are already set up perfectly by asserting that `simple_content_access?` was called as a result of `@can_toggle_sca` being true